### ISSUE TITLE
can't read the data "grid_feature_X101_train.hdf5"

### ISSUE THREAD

**Issue by kelikeli** [lixiangpengcs/CARE#7]
I get a "MemoryError" error when I run the test_grid_feat.py file. I can't read the data, how can I solve this issue? 
This h5 file is only 2.9G

**Comment by ning-mi** [comment]
i ran into this problem too, you have to change the code  which using the .h5 file

**Comment by kelikeli** [comment]
> i ran into this problem too, you have to change the code which using the .h5 file

Wow, thank you very much for your reply, can I ask you how to modify this part of the code to read this h5 file successfully?

**Comment by ning-mi** [comment]
You can try to modify the line 160 of the dataset.py. In this line, the program will create a numpy array to save the features, i think this is reason of the memory error. So i changed it as`        if hparams.grid_feat_trigger:
            with h5py.File(hparams.grid_feature_h5 % self._split, "r") as grid_hdf:
                self.grid_features = grid_hdf['grid_features']
` . Also in the __getitem__ function , you should change some code corrospondingly.

**Comment by kelikeli** [comment]
Thank you so much for your reply, I will try to change the code.

**Comment by kelikeli** [comment]
> You can try to modify the line 160 of the dataset.py. In this line, the program will create a numpy array to save the features, i think this is reason of the memory error. So i changed it as` if hparams.grid_feat_trigger: with h5py.File(hparams.grid_feature_h5 % self._split, "r") as grid_hdf: self.grid_features = grid_hdf['grid_features']` . Also in the **getitem** function , you should change some code corrospondingly.

**Comment by kelikeli** [comment]
> i ran into this problem too, you have to change the code which using the .h5 file

Hello, I want to ask you about this project? I get an error "train_imgid2clidx.pkl does not exist" during the running process. Don't know if you have encountered this problem, I asked the author in a new Issue but got no response, so I want to ask you for help.
Did the author not make the train_imgid2clidx.pkl file public?

**Comment by ning-mi** [comment]
You can see the closed issue 'dan_faster_rcnn', the project MVAN provides the file you need. Actually the CARE is implemented based on MVAN. The region feature is extracted by another project it mentioned, but when i got something wrong with the extracted features and i can't fix it.

**Comment by kelikeli** [comment]
> You can see the closed issue 'dan_faster_rcnn', the project MVAN provides the file you need. Actually the CARE is implemented based on MVAN. The region feature is extracted by another project it mentioned, but when i got something wrong with the extracted features and i can't fix it.

Ok, thanks a lot for your reply, I'm going to look for this file now. Thanks again, I hope that I can run this code as soon as possible.

**Comment by ning-mi** [comment]
good luck!

**Comment by ning-mi** [comment]
> > You can see the closed issue 'dan_faster_rcnn', the project MVAN provides the file you need. Actually the CARE is implemented based on MVAN. The region feature is extracted by another project it mentioned, but when i got something wrong with the extracted features and i can't fix it.
> 
> Ok, thanks a lot for your reply, I'm going to look for this file now. Thanks again, I hope that I can run this code as soon as possible.

Hi, i was wondering if you solve the dataset problem? If you do, would you mind to share the solution to me? Thanks a lot!

**Comment by kelikeli** [comment]
> "train_imgid2clidx.pkl

I'm sorry, but I just found the train_imgid2idx.pkl file, the train_imgid2clidx.pkl is still not found, so the CARE code has not been running. Have you never run this project successfully?

> > > You can see the closed issue 'dan_faster_rcnn', the project MVAN provides the file you need. Actually the CARE is implemented based on MVAN. The region feature is extracted by another project it mentioned, but when i got something wrong with the extracted features and i can't fix it.
> > 
> > 
> > Ok, thanks a lot for your reply, I'm going to look for this file now. Thanks again, I hope that I can run this code as soon as possible.
> 
> Hi, i was wondering if you solve the dataset problem? If you do, would you mind to share the solution to me? Thanks a lot!

I'm sorry, but I just found the train_imgid2idx.pkl file, the train_imgid2clidx.pkl is still not found, so the CARE code has not been running. Have you never run this project successfully?

**Comment by ning-mi** [comment]
> 
I  can't run it because i can't find the correct dataset of grid-feats. I have tried for about a month then i quit!  Wo Ma Le!

**Comment by kelikeli** [comment]
> > 
> 
> I can't run it because i can't find the correct dataset of grid-feats. I have tried for about a month then i quit! Wo Ma Le!

when I run this project, some files do not exist are prompted, but these file authors are indeed not public, have you tried to email the author?

**Comment by ning-mi** [comment]
Actually the author provided some files once, the link of the baiduyun disk is outdate.

**Comment by kelikeli** [comment]
> Actually the author provided some files once, the link of the baiduyun disk is outdate.

Yes, But the files in the link are not enough, there are still files that are missing.

**Comment by ning-mi** [comment]
> 
我给他发过邮件，但他没回。麻了。我搞这个数据集搞了一个多月，下载就花了几个星期。

**Comment by kelikeli** [comment]
> > 
> 
> 我给他发过邮件，但他没回。麻了。我搞这个数据集搞了一个多月，下载就花了几个星期。

我也发邮件了，也没回。。。。。

**Comment by ning-mi** [comment]
> > > 
> > 
> > 
> > 我给他发过邮件，但他没回。麻了。我搞这个数据集搞了一个多月，下载就花了几个星期。
> 
> 我也发邮件了，也没回。。。。。

我真的服了，这个作者啥都不说，而且他论文里给的实验效果很好，特别是生成式的
